saleor/checkout/calculations.py

```diff
@@ -98,17 +154,239 @@ def checkout_line_total(
     lines: Iterable["CheckoutLineInfo"],
     checkout_line_info: "CheckoutLineInfo",
     discounts: Iterable[DiscountInfo] = [],
-) -> "CheckoutTaxedPricesData":
+) -> TaxedMoney:
     """Return the total price of provided line, taxes included.
 
+    It takes in account all plugins.
+    """
+    currency = checkout_info.checkout.currency
+    address = checkout_info.shipping_address or checkout_info.billing_address
```

Comment:
Do we handle the case when the address is not set? In "simple taxes" I use [this logic to pick an address for tax calculations](https://github.com/saleor/saleor/blob/ab8715d81c95dd4c7385b1681ccff7342d5d2467/saleor/tax/utils.py#L20) (agreed with @patrys).

Reply:
In case of tax by the app. We send actual checkout payload to app and App decide how to calculate taxes based on the payload.